cli_tools/gce_vm_image_import/importer/process_planner.go

```diff
@@ -0,0 +1,123 @@
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package importer
+
+import (
+	"errors"
+	"log"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/disk"
+	daisy_utils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	"github.com/GoogleCloudPlatform/compute-image-tools/proto/go/pb"
+	"google.golang.org/api/compute/v1"
+)
+
+// processPlanner determines which actions need to be completed to during processing
+// to ensure a disk is bootable on GCE.
+type processPlanner interface {
+	plan(pd persistentDisk) (*processingPlan, error)
+}
+
+// newProcessPlanner returns a processPlanner that uses disk.Inspector to backfill
+// information that is missing from the arguments (such as: which  OS is on the disk, or whether
+// UEFI is required to boot the disk).
+func newProcessPlanner(args ImportArguments, diskInspector disk.Inspector) processPlanner {
+	return &defaultPlanner{args, diskInspector}
+}
+
+// processingPlan describes the metadata and translation steps that need to be performed
+// against a disk to make it bootable on GCE.
+type processingPlan struct {
+	requiredLicenses        []string
+	requiredFeatures        []*compute.GuestOsFeature
+	translationWorkflowPath string
+}
+
+// metadataChangesRequired returns whether metadata needs to be updated on the
+// GCE disk resource object.
+func (plan *processingPlan) metadataChangesRequired() bool {
+	return len(plan.requiredLicenses) > 0 || len(plan.requiredFeatures) > 0
+}
+
+type defaultPlanner struct {
+	args          ImportArguments
+	diskInspector disk.Inspector
+}
+
+func (p *defaultPlanner) plan(pd persistentDisk) (*processingPlan, error) {
+	// Don't run inspection if the user specified a custom workflow.
+	if p.args.CustomWorkflow != "" {
+		return &processingPlan{translationWorkflowPath: p.args.CustomWorkflow}, nil
+	}
+
+	inspectionResults, inspectionError := p.inspectDisk(pd.uri)
+
+	osID := p.args.OS
+	requiresUEFI := p.args.UefiCompatible
+	if inspectionError == nil && inspectionResults != nil {
+		if osID == "" && inspectionResults.OsCount == 1 && inspectionResults.GetOsRelease() != nil {
+			osID = inspectionResults.GetOsRelease().CliFormatted
+			if p.args.BYOL {
+				osID += "-byol"
+			}
+		}
+
+		if !requiresUEFI {
+			hybridGPTBootable := inspectionResults.GetUefiBootable() && inspectionResults.GetBiosBootable()
+			if !p.args.UefiCompatible && hybridGPTBootable {
```

Comment:
requiresUEFI has the value of p.args.UefiCompatible but both are references here, which can be confusing. Pick one or the other.

Reply:
good find -- done